rotkehlchen/chain/ethereum/interfaces/ammswap/ammswap.py

```diff
@@ -0,0 +1,991 @@
+        if swaps[0].amount0_in > ZERO and swaps[0].amount1_in > ZERO:
+            both_in = True
+        if swaps[-1].amount0_out > ZERO and swaps[-1].amount1_out > ZERO:
+            both_out = True
+
+        if both_in:
+            quote_assets = [
+                (swaps[0].token0, swaps[0].amount0_in if not both_out else swaps[0].amount0_in / 2),  # noqa: E501
+                (swaps[0].token1, swaps[0].amount1_in if not both_out else swaps[0].amount1_in / 2),  # noqa: E501
+            ]
+        elif swaps[0].amount0_in > ZERO:
+            quote_assets = [(swaps[0].token0, swaps[0].amount0_in)]
+        else:
+            quote_assets = [(swaps[0].token1, swaps[0].amount1_in)]
+
+        trade_index = 0
+        if swaps[-1].amount0_out > ZERO:
+            trades = add_trades_from_swaps(
+                swaps=swaps,
+                trades=trades,
+                both_in=both_in,
+                quote_assets=quote_assets,
+                token_amount=swaps[-1].amount0_out,
+                token=swaps[-1].token0,
+                trade_index=trade_index,
+            )
+            trade_index += len(trades)
+        if swaps[-1].amount1_out > ZERO:
+            trades = add_trades_from_swaps(
+                swaps=swaps,
+                trades=trades,
+                both_in=both_in,
+                quote_assets=quote_assets,
+                token_amount=swaps[-1].amount1_out,
+                token=swaps[-1].token1,
+                trade_index=trade_index,
+            )
+
+        return trades
+
+    @staticmethod
+    def swaps_to_trades(swaps: List[AMMSwap]) -> List[AMMTrade]:
+        trades: List[AMMTrade] = []
+        if not swaps:
+            return trades
+
+        # sort by timestamp and then by log index
+        swaps.sort(key=lambda trade: (trade.timestamp, -trade.log_index), reverse=True)
+        last_tx_hash = swaps[0].tx_hash
+        current_swaps: List[AMMSwap] = []
+        for swap in swaps:
+            if swap.tx_hash != last_tx_hash:
+                trades.extend(AMMSwapPlatform._tx_swaps_to_trades(current_swaps))
+                current_swaps = []
+
+            current_swaps.append(swap)
+            last_tx_hash = swap.tx_hash
+
+        if len(current_swaps) != 0:
+            trades.extend(AMMSwapPlatform._tx_swaps_to_trades(current_swaps))
+        return trades
+
+    @staticmethod
+    def _update_assets_prices_in_address_balances(
+            address_balances: AddressToLPBalances,
+            known_asset_price: AssetToPrice,
+            unknown_asset_price: AssetToPrice,
+    ) -> None:
+        """Update the pools underlying assets prices in USD (prices obtained
+        via Inquirer and the subgraph)
+        """
+        for lps in address_balances.values():
+            for lp in lps:
+                # Try to get price from either known or unknown asset price.
+                # Otherwise keep existing price (zero)
+                total_user_balance = FVal(0)
```

Comment:
nitpick: Instead of `FVal(0)` use the `ZERO` constant defined in constants/misc.py